### PR TITLE
Use go modules for vendoring

### PR DESCRIPTION
**Pull request by mikkeloscar**
This PR is to play around with [Go Modules](https://github.com/golang/go/wiki/Modules) which we may want to switch to when Go 1.11 lands.

### CODE REVIEW

File: channel/directory_test.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ func TestDirectoryChannel(t *testing.T) {
 	d := NewDirectory(location)
 	channels, err := d.Update(logger)
 	require.NoError(t, err)
-	require.NotEmpty(t, channels)
+	require.Empty(t, channels)
```
Comment (aermakov-zalando) [inline channel/directory_test.go]: Why?

File: pkg/updatestrategy/node_pool_manager.go [hunk 2]
```diff
@@ -208,7 +208,7 @@ func (m *KubernetesNodePoolManager) taintNode(node *Node, taintKey, taintValue s
 		return nil
 	}
 
-	backoffCfg := backoff.WithMaxTries(backoff.NewConstantBackOff(1*time.Second), maxConflictRetries)
+	backoffCfg := backoff.WithMaxRetries(backoff.NewConstantBackOff(1*time.Second), maxConflictRetries)
```
Comment (szuecs) [inline pkg/updatestrategy/node_pool_manager.go]: Are these changes triggered by version updates of dependencies?

Comment (mikkeloscar) [reply pkg/updatestrategy/node_pool_manager.go]: Yes, sorry, I'll do a separate PR with dependency updates to not have everything included here.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

**Comment by aermakov-zalando** [comment]
Could you pin `k8s.io/api` to a specific version?

**Comment by mikkeloscar** [comment]
@aermakov-zalando the tags used in `k8s.io/api` are not semver which is required by Go modules. Instead it resolves the tag to the commit and uses that instead. I have now pinned it to the commit which is also used in the version of [client-go that we import](https://github.com/kubernetes/client-go/blob/7d04d0e2a0a1a4d4a1cd6baa432a2301492e4e65/Godeps/Godeps.json#L275).

**Comment by mikkeloscar** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
@mikkeloscar oh well :(

**Comment by aermakov-zalando** [comment]
👍

**Pull request merged**